src/Renci.SshNet/SftpClient.cs

```diff
@@ -931,10 +936,15 @@ public IAsyncResult BeginUploadFile(Stream input, string path, bool canOverride,
 
             var flags = Flags.Write | Flags.Truncate;
 
-            if (canOverride)
-                flags |= Flags.CreateNewOrOpen;
+            if (input.Position > 0)
+                flags = Flags.Write | Flags.Append;
```

Comment:
File upload: If the local stream position is not zero, then open the remote file in APPEND mode